examples/routing/src/lib.rs

```diff
@@ -0,0 +1,131 @@
+#[macro_use]
+extern crate log;
+#[macro_use]
+extern crate serde_derive;
+extern crate serde;
+#[macro_use]
+extern crate yew;
+extern crate stdweb;
+
+mod router;
+use router::Route;
+use yew::prelude::*;
+
+
+pub enum Child {
+    A,
+    B,
+    PathNotFound(String)
+}
+
+pub struct Model {
+    child: Child,
+    router: Box<Bridge<router::Router<()>>>
+}
+
+pub enum Msg {
+    NavigateTo(Child),
+    HandleRoute(Route<()>)
+}
+
+impl Component for Model {
+    type Message = Msg;
+    type Properties = ();
+
+    fn create(_: Self::Properties, link: ComponentLink<Self>) -> Self {
+
+        let callback = link.send_back(|route: Route<()>| Msg::HandleRoute(route));
+        let router = router::Router::bridge(callback);
+
+        // TODO Not sure if this is technically correct. This should be sent _after_ the component has been created.
+        // I think the `Component` trait should have a hook called `on_mount()`
+        // that is called after the component has been attached to the vdom.
+        // It seems like this only works because the JS engine decides to activate the
+        // router worker logic after the mounting has finished.
+        router.send(router::Request::GetCurrentRoute);
```

Comment:
I would like to bring a concern about this line to light.

If I understand correctly:
Because the Model hasn't been attached to the VDom yet, if this message was processed by the actor and returned before the create() function completes and the `Model` is mounted, there wouldn't be anything to handle the message returned from the actor, and the message presumably would be lost.

I think that adding a function with and empty default implementation that can be overridden called something like `on_mount(&mut self)` that will be called after the component mounts.